src/Gui/DlgGeneralImp.cpp

```diff
@@ -129,19 +129,21 @@ void DlgGeneralImp::setLanguage()
     if (current != language) {
         hGrp->SetASCII("Language", current.constData());
         Translator::instance()->activateLanguage(current.constData());
+        return true;
     }
+    return false;
 }
 
-void DlgGeneralImp::setNumberLocale()
+void DlgGeneralImp::setNumberLocale(bool force/* = false*/)
 {
     int localeFormat = ui->UseLocaleFormatting->currentIndex();
 
     // Only make the change if locale setting has changed
-    if (localeIndex == localeFormat)
+    if (!force && localeIndex == localeFormat)

```

Comment:
The idea of the class member localeIndex was that if the language has been changed but the locale is still "Operating system" it should avoid to run Translator::instance()->setLocale(). This is because after all the issues we had with InputField we cannot be 100% sure that calling it causes some unknown side-effect.

Reply:
OK, I see. Will have a look now.

Reply:
@wwmayer Should be fixed now